File: app/src/main/java/ave/mujica/poc/pocs/bglogdemo/DemoBgLogTask.java
package ave.mujica.poc.pocs.bglogdemo;

import java.text.SimpleDateFormat;
import java.util.*;
import android.content.Context;
import ave.mujica.poc.bglog.*;

public final class DemoBgLogTask implements BackgroundLogTask {
    public static final String TASK_ID = "bglogdemo.DemoBgLogTask";
    private static final String LOG_TAG = "DemoBgLog";

    @Override
    public String getId() {
        return TASK_ID;
    }

    @Override
    public String getLogTag() {
        return LOG_TAG;
    }

    @Override
    public String getDisplayName() {
        return "Demo Background Log Task";
    }

    @Override
    public long getDefaultIntervalMs() {
        return 1000L;
    }

    @Override
    public BackgroundLogResult runOnce(Context context) throws Exception {
        
        String timestamp = new SimpleDateFormat("HH:mm:ss:SSS", Locale.US).format(new Date());

        List<String> logLines = new ArrayList<>();
        logLines.add(timestamp);
        String notificationText = timestamp;
        String notificationBigText = timestamp;;

        return new BackgroundLogResult(
                getDisplayName(),
                notificationText,
                notificationBigText,
                logLines
        );
    }
}

File: app/src/main/java/ave/mujica/poc/pocs/bglogdemo/DemoBgLogTask.kt
package ave.mujica.poc.pocs.bglogdemo

import android.content.Context

import java.text.SimpleDateFormat
import java.util.Date
import java.util.Locale

import ave.mujica.poc.bglog.BackgroundLogResult
import ave.mujica.poc.bglog.BackgroundLogTask


class DemoBgLogTask : BackgroundLogTask {

    companion object {
        const val TASK_ID = "bglogdemo.DemoBgLogTask"
        private const val LOG_TAG = "DemoBgLog"
    }

    override fun getId(): String {
        return TASK_ID
    }

    override fun getLogTag(): String {
        return LOG_TAG
    }

    override fun getDisplayName(): String {
        return "Demo Background Log Task"
    }

    override fun getDefaultIntervalMs(): Long {
        return 1000L
    }

    override fun runOnce(context: Context): BackgroundLogResult {
        val timestamp = SimpleDateFormat("HH:mm:ss:SSS", Locale.US).format(Date())

        val logLines = arrayListOf<String>()
        logLines.add(timestamp)

        return BackgroundLogResult(
            getDisplayName(),
            timestamp,
            timestamp,
            logLines
        )
    }
}